### DISCUSSION TITLE
Tweak: replaceInternalLinks(node) --- Convert Internal Links to <a>tag and replace .md to .html

### DISCUSSION THREAD

**Discussion by vgyenge6** [mvdkwast/obsidian-copy-as-html#62] [Ideas]
Hi,

I little tweaked your main.js because I found a similar post https://github.com/mvdkwast/obsidian-copy-as-html/issues/47#issue-2000448338 and I did it because I had also a similar problem with an urgent need.

However, I did not wish to use the Obsidian's URL of the file instead of changing the .md to .html given in the original 
 ```
 <a class="internal-link" href="filename.md" data-href="filename.md" target="_blank"
  
or

 <a class="internal-link" href="filename.md#anchor" data-href="filename.md#anchor" target="_blank"
 
to achieve a result

 <a class="internal-link" href="filename.html" data-href="filename.html" target="_self"
 
 or
 
 <a class="internal-link" href="filename.html#anchor" data-href="filename.html#anchor" target="_self"
 ```
 The target="_self" is needed for me because I am about to use the generated HTML file in an iframe, thus the navigation needs to remain in the iframe. 
 
 Thus I changed the replaceInternalLinks(node) function in your main.js
 
 ```
 replaceInternalLinks(node) {
    node.querySelectorAll("a.internal-link").forEach((node2) => {

		var t = node2.getAttribute("href").replace(/\.md/, ".html");
		var t1 = node2.getAttribute("data-href").replace(/\.md/, ".html");

		node2.setAttribute("href", t)
		node2.setAttribute("data-href", t1);
		node2.setAttribute("target", "_self");
    }); 
  }
	replaceInternalTagLinks(node) {
    node.querySelectorAll("a.tag").forEach((node2) => {
		node2.setAttribute("target", "_self");
    }); 
  }

I look forward to hearing your opinion. Thank you.

**Comment by mvdkwast** [comment]
Hi again :)

Thanks for reporting. This request clearly has a multi-document export flair to it, and that is somewhere where I didn't want to go. The original scope of this plugin was really to ease sharing Obsidian notes through mail/documents, and not to export multiple documents to HTML, and I'm afraid this would only be a first step to such a feature.

Now I don't say that this isn't a useful (although there are already a few plugins that do this), but it's clearly outside the time-budget I have for this and my initial goals. 

Sorry :'(

**Reply by mvdkwast** [reply]
Finally decided to include this in dab43615e7aae904e462ce84a3e5fedb86fa8219

**Reply by vgyenge6** [reply]
Hi,

Thank you for dealing with and working on this issue and in general my other former ones (#58, #63) and finally resolving and closing them.

I momentarily do have not enough time to check and try the newer versions, but if I have I will come back.

Further good work for you.
